doc/source/serve/http-guide.md

```diff
@@ -95,88 +117,148 @@ def f():
 class FastAPIWrapper:
     pass
 
-FastAPIWrapper.deploy()
+serve.run(FastAPIWrapper.bind())
 ```
 
 This is useful for scaling out an existing FastAPI app with no modifications necessary.
 Existing middlewares, automatic OpenAPI documentation generation, and other advanced FastAPI features should work as-is.
-You can also combine routes defined this way with routes defined on the deployment:
+
+To try it out, save a code snippet in a local python file (i.e. main.py) and in the same directory, run the following commands to start a local Ray cluster on your machine.
+
+```bash
+ray start --head
+python main.py
+```
+
+(serve-http-adapters)=
+
+## HTTP Adapters
+
+HTTP adapters are functions that convert raw HTTP requests to basic Python types that you know and recognize.
+
+For example, here is an adapter that extracts the JSON content from a request:
 
 ```python
-import ray
+async def json_resolver(request: starlette.requests.Request):
+    return await request.json()
+```
 
-from fastapi import FastAPI
-from ray import serve
+The input arguments to an HTTP adapter should be type-annotated. At a minimum, the adapter should accept a `starlette.requests.Request` type (https://www.starlette.io/requests/#request),
+but it can also accept any type that's recognized by [FastAPI's dependency injection framework](https://fastapi.tiangolo.com/tutorial/dependencies/).
 
-app = FastAPI()
-ray.init(address="auto", namespace="summarizer")
-serve.start(detached=True)
+Here is an HTTP adapter that accepts two HTTP query parameters:
 
-@app.get("/")
-def f():
-    return "Hello from the root!"
+```python
+def parse_query_args(field_a: int, field_b: str):
+    return YourDataClass(field_a, field_b)
+```
 
-@serve.deployment(route_prefix="/api1")
-@serve.ingress(app)
-class FastAPIWrapper1:
-    @app.get("/subpath")
-    def method(self):
-        return "Hello 1!"
+You can specify different type signatures to facilitate the extraction of HTTP fields, including
+- [query parameters](https://fastapi.tiangolo.com/tutorial/query-params/),
+- [body parameters](https://fastapi.tiangolo.com/tutorial/body/),
+and 
+- [many other data types](https://fastapi.tiangolo.com/tutorial/extra-data-types/).
 
-@serve.deployment(route_prefix="/api2")
-@serve.ingress(app)
-class FastAPIWrapper2:
-    @app.get("/subpath")
-    def method(self):
-        return "Hello 2!"
+For more details, you can take a look at the [FastAPI documentation](https://fastapi.tiangolo.com/).
+
+You can use adapters in different scenarios within Serve, which we will go over one by one:
+
```

Comment:
May be make a transition here. 

In addition to above adapters, you also use other adapters. Below we examine at least three: